specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/serverVulnerabilityAssessments.json

```diff
@@ -0,0 +1,338 @@
+                        }
+                    }
+                }
+            },
+            "put": {
+                "x-ms-examples": {
+                    "Create a server vulnerability assessments on a resource. Only 'default' resource is supported. Once creating the resource, the server will be onboarded to vulnerability assessment by Microsoft.Security": {
+                        "$ref": "./examples/ServerVulnerabilityAssessments/CreateServerVulnerabilityAssessments_example.json"
+                    }
+                },
+                "description": "Creating a server evulnerability assessment on a resource, which will onboard a resource for having a vulnerability assessment on it",
+                "tags": [
+                    "ServerVulnerabilityAssessment"
+                ],
+                "operationId": "ServerVulnerabilityAssessment_CreateOrUpdate",
+                "parameters": [
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+                    },
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupName"
+                    },
+                    {
+                        "name": "resourceNamespace",
+                        "in": "path",
+                        "description": "The Namespace of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "resourceType",
+                        "in": "path",
+                        "description": "The type of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "resourceName",
+                        "in": "path",
+                        "description": "Name of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "serverVulnerabilityAssessment",
+                        "in": "path",
+                        "description": "ServerVulnerabilityAssessment status. only a 'default' value is supported.",
+                        "required": true,
+                        "type": "string",
+                        "enum": ["default"]
+                    },
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/ServerVulnerabilityAssessment"
+                        }
+                    },
```

Comment:
Based on the modeling of the payload and the examples, the PUT appears to be a long running asynchronous operation.

If this is indeed the case, it looks like you should instead of implement one of the patterns documented at ([https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations)). The closest standard equivalent to what you have here is the provisioningState pattern.